### ISSUE TITLE
Capture previous values when changes occur

### ISSUE THREAD

**Issue by westonruter** [wp-stream/stream#71]
Whenever a user action results in a change to data in WordPress, the original value could be stored along with the stream entry. This would allow the previous value to be compared with the new value. If a menu item was deleted, this would then make possible the restoration of that item. This would not be necessary to store old data for post revisions since that is a builtin feature, but it would be useful for tracking changes to postmeta, terms, menus, widgets, etc. There are obvious integration points with Settings Revisions here.

Basically what we're looking at here is using Stream as a version control system.
